### ISSUE TITLE
lintGradle unusable output / generateGradleLintReport no output

### ISSUE THREAD

**Issue by thahnen** [nebula-plugins/gradle-lint-plugin#343]
We are using the newest nebula.lint version (17.0.0) with the current Gradle 7.1 version at our company to lint our giant monolith of a software. It consists of a multi-project build with at least 25 subprojects.

In the top most build.gradle file (root project) we followed the documentation to include nebula.lint using the following (shortend) scheme:
```gradle
plugins {
    id 'nebula.lint' version '17.0.0' apply(false)
}

// ...

allprojects {
    apply plugin: 'nebula.lint'
    gradleLint {
        alwaysRun = false
        rules = ['unused-dependency']
    }
}

// ...
```

When running the *lintGradle* task which is only available at root project level, we got some useful output of unused dependencies (very neat!) but also something like this:
> warning   unused-dependency                  one or more classes in jakarta.xml.bind:jakarta.xml.bind-api:2.3.2 are required by your code directly (no auto-fix available)

And yes we did get more warnings than one (this project is enormous). The problem are as follows:
1. *no auto-fix available* -> sad but this plugin has its limits I understand, but ...
2. ... there is no information on which subproject this warning occured!

And with over 100 warnings like this we cannot check every subproject manually, this is crazy! Also this warning is quite confusing. Does it mean, the error occured because of a wrong version (we got some for junit:junit:4.13 and 4.11 while only using 4.13) or transitive includes? Is it possible to use another rule to find more information (while there is no list about all rules available sadly)?

I felt a bit bamboozled when running lintGradle providing to usable output on this warnings. But I also found no option to turn this particular warning off, maybe I missed something. I would be happy only getting the messages about actually unused dependencies which could be simply removed (like a trigger to turn off everything containing *no auto-fix available*)!

Earlier when some other warning were triggered including some real unused dependencies which I later removed I tried to use *generateGradleLintReport* to create such a report.
I ran this task in the root project after running *lintGradle*, nothing skipped everything was fine. But the report was missing and no error thrown. Also running *generateGradleLintReport* with "--info" did not show anything.

I think this task is broken. While available in subprojects running it also provided no output at all (nowhere) :(

**Comment by mathroule** [comment]
After upgrading Gradle from version 7.0.2 to 7.1.1 and Gradle lint plugin from version 16.23.0 to 17.0.0 we have the same kind of issue.
The report are not generated anymore.

**Comment by nkiesel** [comment]
Same here: `lintGrade` reports "missing direct dependencies" but not from which module, and `generateGradleLintReport` produces per-module reports but these are empty.
